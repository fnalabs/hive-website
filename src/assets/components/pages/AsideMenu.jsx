import React from 'react'
import { NavLink } from 'react-router-dom'

import { AnalyticsLink } from 'common'

const AsideMenu = () => (
  <aside className='menu'>
    <p className='menu-label has-text-weight-bold'>v2.1.3</p>
    <ul className='menu-list'>
      <li><NavLink to='/overview' activeClassName='is-active'>Overview</NavLink>
        <ul>
          <li><NavLink to='/model' activeClassName='is-active'>Data Model</NavLink></li>
          <li><NavLink to='/domain' activeClassName='is-active'>Domain Logic</NavLink></li>
          <li><NavLink to='/infrastructure' activeClassName='is-active'>Infrastructure</NavLink></li>
          <li><NavLink to='/telemetry' activeClassName='is-active'>Telemetry</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/start' activeClassName='is-active'>Get Started</NavLink>
        <ul>
          <li><NavLink to='/setup' activeClassName='is-active'>Setup</NavLink></li>
          <li className='menu-label has-text-weight-bold'>Examples:</li>
          <li><NavLink to='/basic' activeClassName='is-active'>Basic</NavLink></li>
          <li><NavLink to='/rest' activeClassName='is-active'>Rest</NavLink></li>
          <li><NavLink to='/cqrs-es' activeClassName='is-active'>CQRS/ES</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/documentation' activeClassName='is-active'>Documentation</NavLink>
        <ul>
          <li><NavLink to='/environments' activeClassName='is-active'>Environments</NavLink></li>
          <li><AnalyticsLink to='https://fnalabs.github.io/hive-io/' target='_blank' rel='noopener noreferrer'>API</AnalyticsLink></li>
        </ul>
      </li>
    </ul>
  </aside>
)

export default AsideMenu
