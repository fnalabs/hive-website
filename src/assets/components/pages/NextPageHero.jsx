import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { AnalyticsLink } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { ArrowRight } from 'icons'

const toRegex = /^https:\/\//

export default class NextPageHero extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    const to = this.props.to

    return (
      <Hero size='small' className='is-hive'>
        <HeroBody centered>
          {toRegex.test(to)
            ? <AnalyticsLink to={to}><ArrowRight height='48' /></AnalyticsLink>
            : <Link to={to}><ArrowRight height='48' /></Link>}
        </HeroBody>
      </Hero>
    )
  }
}
