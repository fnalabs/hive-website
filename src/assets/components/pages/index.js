// site's main pages
export { default as Home } from './home/Home.jsx'
export { default as Overview } from './overview/Overview.jsx'
export { default as Start } from './start/Start.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
