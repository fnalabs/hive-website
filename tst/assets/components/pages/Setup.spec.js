/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { MemoryRouter } from 'react-router'
import ReactGA from 'react-ga'

import { Setup } from 'pages'

jest.mock('react-ga')

describe('<Setup />', () => {
  it('should render Setup page static content', () => {
    const tree = renderer.create(<MemoryRouter><Setup location={{ pathname: '/domain' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render Setup page static content and report page view', () => {
    shallow(<Setup location={{ pathname: '/domain' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
