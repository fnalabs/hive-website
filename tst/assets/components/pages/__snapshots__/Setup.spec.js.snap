// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Setup /> should render Setup page static content 1`] = `
Array [
  <article
    className="section is-medium"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <header
          className="column is-narrow is-hidden-touch"
        >
          <aside
            className="menu"
          >
            <p
              className="menu-label has-text-weight-bold"
            >
              v2.1.2
            </p>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  aria-current={null}
                  href="/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/model"
                      onClick={[Function]}
                    >
                      Data Model
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/domain"
                      onClick={[Function]}
                    >
                      Domain Logic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/infrastructure"
                      onClick={[Function]}
                    >
                      Infrastructure
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/telemetry"
                      onClick={[Function]}
                    >
                      Telemetry
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/start"
                  onClick={[Function]}
                >
                  Get Started
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/setup"
                      onClick={[Function]}
                    >
                      Setup
                    </a>
                  </li>
                  <li
                    className="menu-label has-text-weight-bold"
                  >
                    Examples:
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/basic"
                      onClick={[Function]}
                    >
                      Basic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/rest"
                      onClick={[Function]}
                    >
                      Rest
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/cqrs-es"
                      onClick={[Function]}
                    >
                      CQRS/ES
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/documentation"
                  onClick={[Function]}
                >
                  Documentation
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/environments"
                      onClick={[Function]}
                    >
                      Environments
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      API
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </aside>
        </header>
        <section
          className="column content"
          role="document"
        >
          <h1>
            Setup
          </h1>
          <p>
            Hive
            <sup>
              io
            </sup>
             has minimal requirements for installation, allowing you to choose the solutions that suits your needs regarding storage, security, monitoring, and more. Once you have your cloud platform decided, you can start to code your domain logic that will eventually be deployed with one of the container types.
          </p>
          <h2
            id="domain_logic"
          >
            Domain Logic
          </h2>
          <p>
            To get started with implementing your domain logic, install the following NPM package to your domain module(s). Other dependencies of your domain logic for your microservices will need to be installed separately.
          </p>
          <p>
            <pre>
              $ npm install --save hive-io
            </pre>
          </p>
          <h2
            id="infrastructure"
          >
            Infrastructure
          </h2>
          <p>
            There are currently 4 infrastructure images available to cover anything from basic to complex microservice needs.
          </p>
          <dl>
            <dt>
              <a
                href="https://hub.docker.com/r/fnalabs/hive-base-js/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <em>
                  <strong>
                    Base
                  </strong>
                </em>
              </a>
            </dt>
            <dd>
              <pre>
                $ docker pull fnalabs/hive-base-js:&lt;[release]|latest&gt;
              </pre>
            </dd>
            <dt>
              <a
                href="https://hub.docker.com/r/fnalabs/hive-producer-js/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <em>
                  <strong>
                    CQRS/ES Producer
                  </strong>
                </em>
              </a>
            </dt>
            <dd>
              <pre>
                $ docker pull fnalabs/hive-producer-js:&lt;[release]|latest&gt;
              </pre>
            </dd>
            <dt>
              <a
                href="https://hub.docker.com/r/fnalabs/hive-consumer-js/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <em>
                  <strong>
                    CQRS/ES Consumer
                  </strong>
                </em>
              </a>
            </dt>
            <dd>
              <pre>
                $ docker pull fnalabs/hive-consumer-js:&lt;[release]|latest&gt;
              </pre>
            </dd>
            <dt>
              <a
                href="https://hub.docker.com/r/fnalabs/hive-stream-processor-js/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <em>
                  <strong>
                    CQRS/ES Stream Processor
                  </strong>
                </em>
              </a>
            </dt>
            <dd>
              <pre>
                $ docker pull fnalabs/hive-stream-processor-js:&lt;[release]|latest&gt;
              </pre>
            </dd>
          </dl>
          <p>
            So far, the list above defines some common types of microservices you may need. Future versions of the framework will add more types to the above and add support for multiple languages.
          </p>
        </section>
      </div>
    </div>
  </article>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="previous page"
            href="/start"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="/basic"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
